src/ec/packages/ec-component-hero-banner/hero-banner.story.js

```diff
@@ -14,107 +15,94 @@ import dataLeft from './demo/data--align-left';
 import heroBanner from './ecl-hero-banner.html.twig';
 import notes from './README.md';
 
-const PrepareBanner = data => {
+const prepareBanner = data => {
+  if (data.title) {
+    data.title = text('title', data.title, buttonLabels.required);
+  }
+
+  if (data.link.link.label) {
+    data.link.link.label = text(
+      'link.Link.label',
+      data.link.link.label,
+      buttonLabels.required
+    );
+  }
+  if (data.description) {
+    data.description = text(
+      'description',
+      data.description,
+      buttonLabels.required
+    );
+  }
+
+  if (data.centered) {
+    data.centered = boolean('centered', data.centered, buttonLabels.optional);
+  }
+
+  if (data.image) {
+    data.image = text('image', data.image, buttonLabels.optional);
```

Comment:
I don't think this should be "optional", well it is "optional" in the general definition of the component, but since the demo we have for the banner is showing all the different variants including the ones without an image (primary and default) in the "image" variant of the component the image is actually mandatory.
So, it should not be possible to see the "image" variant without an image, the result is wrong, because it should actually fallback to one of the aforementioned ones (primary or default).
Furthermore, i don't see any knob for the fundamental parameter "type". This is an optional parameter (since we have a default value for that) and we should show it in a select element with the current type option only. ;)